### ISSUE TITLE
Suggestion: Active jamming

### ISSUE THREAD

**Issue by ziproot** [endless-sky/endless-sky#10154]
### Problem Description

The cloaking device is a very powerful reward. It is so powerful that it alone is causing people to choose one side of the main plot more than the other. The current solution appears to be to make the cloaking device into an active jammer.

Problem: There is no such functionality in the game.

### Related Issue Links

#3825 - Proposal for making cloaks jammers, but only if you take damage
#4119 - Proposal for ways to toggle cloaking devices, some of which could be combined to make them jammers
#4121 - The main use case for active jamming
#7104 - Proposal for a “stealth” mechanic, of which the cloaking device could tie into in a way that isn’t as powerful as it is now

### Desired Solution

A way to have “active jamming” which involves pressing a key to increase your optical and/or radar jamming at the expense of energy/heat, but only until the key is pressed again. This can allow for more powerful jamming that uses more energy and produces more heat, but most be activated.

### Alternative Approaches

The cloaking device could receive more flexibility as suggested in numerous other issues.

Active jamming could use the same *key* as cloaking, despite having different tags, and cloaking could be renamed to “stealth.” This would also solve the issue of cloaking being “spoiled” for newer players, as #10139 is trying to do.

A more generic system that allows toggling other attributes with a hotkey, to allow making something like an emergency shield generator, for instance.

### Additional Context

The catalyst for this feature request is the cloaking device that you get at the end of free worlds reconciliation, which is very powerful at the moment as it allows you to essentially exit combat whenever you want. This is unnecessary to complete the one mission that it is required in, as just a few cannon fodder ships can get the job done. The cloaking device is overkill for the dodge the nukes mission, and an active jammer would work better, but that would require the possibility of making an active jammer.

**Comment by Zarkonnen** [comment]
I agree a more universal "activate stealth/jamming" system would open a lot of possibilities.

> just a few cannon fodder ships can get the job done

Note that the cannon fodder approach works but feels very out of step with the plot contents. Sacrificing ships full of people is what you're trying to avoid.

> using a tag defining a number (e.g. 1-10) that could be bound in the settings

Unfortunately the number keys are already in use for escort controls. So is ctrl/cmd+number, and I'm hoping to use shift+number for [planet selection](https://github.com/endless-sky/endless-sky/pull/10181).
